src/containers/BooksForm.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions';
+
+const categories = [
+  'Action',
+  'Biography',
+  'History',
+  'Horror',
+  'Kids',
+  'Learning',
+  'Sci-Fi',
+];
+
+const mapStateToProps = ({ books }) => ({ books });
+const mapDispatchToProps = dispatch => ({
+  createBook: book => dispatch(createBook(book)),
+});
+
+const BooksForm = ({ createBook }) => {
+  const [errors, setErrors] = useState({});
+  const [book, setBook] = useState({
+    id: Math.floor(Math.random() * 1000),
```

Comment:
- [ ] This id is being generated when the application starts and remain the same for every newly created book. If you add more than one book you will notice that they are being saved with the same id. What you can do instead is to generate the id at the moment you create a book object. This way, the id does not need to be saved in the state
```javascript
createBook({ id: Math.floor(Math.random() * 1000), title: book.title, category: book.category });
```